sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/parallel/parallel_job.py

```diff
@@ -108,20 +109,58 @@ def _to_component(self, context: Optional[Dict] = None, **kwargs) -> "ParallelCo
         context = context or {BASE_PATH_CONTEXT_KEY: Path("./")}
 
         # Create anonymous parallel component with default version as 1
+        init_kwargs = {}
+        for key in [
+            "mini_batch_size",
+            "partition_keys",
+            "logging_level",
+            "max_concurrency_per_instance",
+            "error_threshold",
+            "mini_batch_error_threshold",
+            "retry_settings",
+            "resources",
+        ]:
+            value = getattr(self, key)
+            from azure.ai.ml.entities import BatchRetrySettings, JobResourceConfiguration
+
+            if key == "retry_settings" and isinstance(value, BatchRetrySettings):
+                values_to_check = [value.max_retries, value.timeout]
+            elif key == "resources" and isinstance(value, JobResourceConfiguration):
+                values_to_check = [
+                    value.locations,
+                    value.instance_count,
+                    value.instance_type,
+                    value.shm_size,
+                    value.max_instance_count,
+                    value.docker_args,
+                ]
+            else:
+                values_to_check = [value]
+
+            # note that component level attributes can not be data binding expressions
+            # so filter out data binding expression properties here;
+            # they will still take effect at node level according to _to_node
+            if any(
```

Comment:
Thinking that will this effect component reuse? As we filter some component settings out..

Reply:
Yes. This is a workaround as such behaviors (the transformation between parallel job and parallel component and reusage between them) hasn't been clearly defined in the begining.

Current expectation is that, a node can be reused if the usage doesn't change (parallel job => parallel job; parallel component => parallel component). PRS team may update this once they clearly defined this.